crates/extensions/tedge_mqtt_ext/src/trie.rs

```diff
@@ -1078,12 +1122,29 @@ mod tests {
                 }
                 for s in subscriptions {
                     let diff = t.remove(&s,&1);
-                    for added in diff.subscribe {
+
+                    for added in &diff.subscribe {
                         // If one subscribes to some topic, then one must unsubscribe from a larger one
-                        assert!(diff.unsubscribe.iter().any(|removed| {
-                            let cmp = RankTopicFilter(removed).partial_cmp(&RankTopicFilter(&added));
-                            cmp == Some(Ordering::Greater)
-                        }));
+                        assert!(
+                            RankTopicFilter(&s) > RankTopicFilter(added),
+                            "Unsubscribing from {s} resubscribed to topic that isn't larger ({added}): {diff:?}");
+                    }
+                }
```

Comment:
I think it would be better to keep symmetric the two properties:

- `mqttrie_extend_is_growing_when_adding_subscriptions`
- `mqttrie_extend_is_decreasing_when_removing_subscriptions`

Reply:
Done: https://github.com/thin-edge/thin-edge.io/pull/3661/commits/1c0ca94201076d862610ca32719c3c66506f1d49